src/vellum/workflows/resolvers/resolver.py

```diff
@@ -16,27 +18,66 @@ def get_latest_execution_events(self) -> Iterator[WorkflowEvent]:
     def get_state_snapshot_history(self) -> Iterator[BaseState]:
         return iter([])
 
-    def load_state(self, previous_execution_id: Optional[UUID] = None) -> Optional[BaseState]:
+    def _find_execution_events(
+        self, execution_id: str, spans: list
+    ) -> Tuple[Optional[WorkflowExecutionInitiatedEvent], Optional[WorkflowExecutionInitiatedEvent]]:
+        previous_invocation: Optional[WorkflowExecutionInitiatedEvent] = None
+        root_invocation: Optional[WorkflowExecutionInitiatedEvent] = None
+
+        for span in spans:
+            # Look for workflow execution spans
+            if isinstance(span, WorkflowExecutionSpan):
+                # Find the WorkflowExecutionInitiatedEvent in the span's events
+                for event in span.events:
+                    if isinstance(event, WorkflowExecutionInitiatedEvent):
+                        # Check if this is the previous execution matches the execution_id
+                        if event.span_id == execution_id:
+                            previous_invocation = event
+
+                        if not event.links:
+                            root_invocation = event
+
+                        if previous_invocation and root_invocation:
+                            break
+
+        return previous_invocation, root_invocation
+
+    def load_state(self, previous_execution_id: Optional[str] = None) -> LoadStateResult:
         if previous_execution_id is None:
-            return None
+            return LoadStateResult(state=None, span_link_info=None)
 
         if not self._context:
             logger.warning("Cannot load state: No workflow context registered")
-            return None
+            return LoadStateResult(state=None, span_link_info=None)
```

Comment:
[nit] not sure of the benefit of a second nested type (span_link_info) and could see just including it all in the root